### PR TITLE
My ip2

### PR DESCRIPTION
**Pull request by barbican-youth**
bastion_current_ip.py has been edited to accept command line options as follows:

  -k SSH_KEY_NAME, --ssh_key_name SSH_KEY_NAME
  -s SUBNET_LIST, --subnet_list SUBNET_LIST
  -n NAME, --name NAME
  -z ZONE_NAME, --zone_name ZONE_NAME
  -r REGION, --region REGION
  --destroy
  --plan_destroy
  --plan_apply
  --apply

@bknelson

### CODE REVIEW

File: bastion_current_ip.py [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+import os,subprocess
+import argparse
+
+
+
+def main():
+
+	parser = argparse.ArgumentParser(description="Pass Terraform variables for bastion")
+
+	parser.add_argument('-k','--ssh_key_name', type=str)
```
Comment (bknelson) [inline bastion_current_ip.py]: For each of these arguments, add a 'help' field

File: bastion_current_ip.py [hunk 2]
```diff
@@ -0,0 +1,76 @@
+# obtain current public ip, create a variables.tfvars file
```
Comment (bknelson) [inline bastion_current_ip.py]: Run this through a linter, so your code is consistant with PEP8 standard.  I suggest pylint.

File: bastion_current_ip.py [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+
+		if args.apply:
+			os.system('./apply_bastion.sh')
+			return
+
+
+def get_public_ip(): #attempt to obtain public ip. tries 2 different websites
+	try:
+		myip = subprocess.check_output(["dig","+short","myip.opendns.com","@resolver1.opendns.com"])
+	except:
```
Comment (bknelson) [inline bastion_current_ip.py]: Can you be more specific about the exceptions you are catching?

File: bastion_current_ip.py [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+		try:
+			myip = subprocess.check_output(["curl", "-s","checkip.dyndns.org"])
+			myip = myip.split(":",1)[-1]
+			myip = myip.split("<",1)[0]
+		except:
```
Comment (bknelson) [inline bastion_current_ip.py]: Can you be more specific about the exceptions you are catching?

File: bastion_current_ip.py [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+def get_public_ip(): #attempt to obtain public ip. tries 2 different websites
+	try:
+		myip = subprocess.check_output(["dig","+short","myip.opendns.com","@resolver1.opendns.com"])
+	except:
+		try:
+			myip = subprocess.check_output(["curl", "-s","checkip.dyndns.org"])
+			myip = myip.split(":",1)[-1]
+			myip = myip.split("<",1)[0]
+		except:  
+			myip = raw_input('Error obtaining IP address. Please specify IP: ')
```
Comment (bknelson) [inline bastion_current_ip.py]: Why not throw an exception with extra info?

**Review by bknelson — CHANGES_REQUESTED** [review]

**Comment by barbican-youth** [comment]
@bknelson  please see latest commit for changes implemented

**Review by bknelson — APPROVED** [review]
:shipit:

**Pull request merged**